cyences_app_for_splunk/bin/sophos_instance_details.py

```diff
@@ -0,0 +1,217 @@
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+
+            elif(hostname!="" and hostname!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?hostnameContains="+hostname, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            return list_of_uuid
+        except Exception as e:
+            self.logger.error("Error while fetching Instance UUID : {}".format(e))
+
+
+
+    def generate(self):
+
+        ip = self.ip
+        hostname = self.hostname
+        uuid = self.uuid
+        all_endpoints = self.all_endpoints
+        if(ip=="" and hostname=="" and uuid=="" and ip==" " and hostname==" " and uuid==" " and all_endpoints==False):
+            self.logger.error("Please provide IP or Hostname or UUID")
+            exit()
+
+        # Read Client ID & Client Secret
+        client_id,client_secret = self.get_client_details()
+
+        if not client_id or not client_secret:
+            self.logger.error("Sophos Client ID or Client Secret or both not found in the cs_configurations.conf file.")
+        else:
+            barier_token = self.get_barier_token(client_id,client_secret)
+            self.get_tenant_list(barier_token)
+
+            if(all_endpoints):
+                for i in SOPHOS_TENANT_DICT:
+
+                    current_page=1
+                    total_page = 1
+                    while(current_page<=total_page):
+                        requestHeaders = {
+                            "X-Tenant-ID": i,
+                            "Authorization": "Bearer "+ barier_token,
+                            "Accept": "application/json"
+                        }
+                        if(current_page==1):
+                            requestUrl = SOPHOS_TENANT_DICT.get(i)+"/endpoint/v1/endpoints?pageTotal=true"
+                        else:
+                            requestUrl =  SOPHOS_TENANT_DICT.get(i)+"/endpoint/v1/endpoints?page="+str(current_page)
+                        request = requests.get(requestUrl, headers=requestHeaders)
+                        if(request.status_code!=200):
+                            self.logger.error("Error while fetching all the endpoints")
+                            exit()
```

Comment:
raise error so that it can get notified in the search UI. If this error is not expected and you don't want to continue the rest of the code, otherwise this is fine.